src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java

```diff
@@ -84,35 +118,360 @@ public void visitToken(DetailAST ast) {
     }
 
     /**
+     * @return an ordered by insertion set of {@link OverloadGroup}.
+     */
+    public String getModifierGroups() {
+        return Objects.toString(overloadGroups, null);
+    }
+
+    /**
+     * Set static methods to be grouped separately from instance methods.
+     * @param modifierGroupRegex true means static method overloads are
+     *                       evaluated separately from instance methods.
+     */
+    public void setModifierGroups(String modifierGroupRegex) {
+        final String safeModifierGroup = Objects.toString(modifierGroupRegex, "");
```

Comment:
@ilanKeshet `null` can't be achieved through a property in our configuration file and passed to the setter. Please show me example where it can be otherwise.